### ISSUE TITLE
Allow to force password changes after the next UI login

### ISSUE THREAD

**Issue by nickvergessen** [nextcloud/password_policy#14]
https://help.nextcloud.com/t/force-password-change-at-user-logon/1664/1

I think it's a nice idea and also fit's in the context of this app.

**Comment by MorrisJobke** [comment]
Moved to 11

**Comment by klm46** [comment]
Hi there. Sorry for my stupid question. This is available @ Nextcloud 11.0. There is no patch for now, right?
Our data protection commissioner stopped  my nextcloud-Installation for going prod. for that reason.

**Comment by nickvergessen** [comment]
That's odd. But no, this is not implemented yet.

**Comment by klm46** [comment]
No - this is Germany ;)

**Comment by LukasReschke** [comment]
Use LDAP login and policies from the LDAP are used.

Also as a company you may consider a Suppodf Subscription from https://nextcloud.com/enterprise. This gives you direct access to Engineers and influence on our roadmap.

**Comment by klm46** [comment]
This is an option for our company, for sure. I did take note of this a few weeks ago and planed to ask for such an invest after the Cloud is productional. 

For LDPA-User it is no Problem. But for Partner Companies we need to create databased Accounts. And the workflow does not have an opportunity to force new database-user to change the PW we created when adding the account. 
Again, the premium Subsribtion is in plan - we have much higher costs for less usefull software ;)

**Comment by LukasReschke** [comment]
> This is an option for our company, for sure. I did take note of this a few weeks ago and planed to ask for such an invest after the Cloud is productional.
> 
> For LDPA-User it is no Problem. But for Partner Companies we need to create databased Accounts. And the workflow does not have an opportunity to force new database-user to change the PW we created when adding the account. 
Again, the premium Subsribtion is in plan - we have much higher costs for less usefull software ;)

Awesome. Reach out and we'd likely be very happy to help with a proof of concept. Just link to this discussion :-)

So the requirement is actually that after registration users have to change their password is that correct? Would it also be an option that if you have created an user a mail with password reset link is send to the user? The user would then have to reset their password there.

Also is there any kind of requirement for changing passwords after X days? That would again be kinda harder to implement and not sure if at the moment desired since it clutters the UI and is not compatible with all backends. (i.e. confusing behaviour)

**Comment by klm46** [comment]
> So the requirement is actually that after registration users have to change their password is that correct? Would it also be an option that if you have created an user a mail with password reset link is send to the user? The user would then have to reset their password there.


This would be an acceptable workaround. Because this would ensure, that no one of us (expecting us admins with database knowledge) knows the PW of the external Partner. No need to force changing PW after X days.

> Awesome. Reach out and we'd likely be very happy to help with a proof of concept. Just link to this discussion :-)

I will do this. But we are a relative big Company (about 1000 Employes) and it takes a while since all the processes being past. But I'll come back to you soon with this.

**Comment by klm46** [comment]
any news about the workaround?

**Comment by kilrau** [comment]
BUMP

**Comment by elpraga** [comment]
Yes, I'm also interested to know if this feature is already available.

**Comment by nsturtz** [comment]
is this feature is available for NC 20.04?

**Comment by nsturtz** [comment]
that is my server version 🤣 I meant 19.0.1?

**Comment by schlagi123** [comment]
I think this is a very important feature because all users without ldap or other environments with there own password policies can not handle the account createn (send password to user without registration app) or setup a expiration period for the password. This two features are basic for me, and I shocked, that this features on the one hand not avilable at the nextcloud and on the other hand this topic was not dealt with for so long, although corresponding issues were created (this issue and https://github.com/nextcloud/server/issues/1262).

I hoped Nextcloud would be the better owncloud!? But owncloud has this features long time implemented (https://doc.owncloud.com/server/10.1/admin_manual/configuration/server/security/password_policy.html) and to me it feels as if nextcloud never wanted to implement this function.

I'm not a php programmer, but I think for the nextcloud experts it is not so hard to implement this functions (you can take a look at ownclouds implementation).

This offends the community, which is also responsible for your success. I would like this function to be implemented soon, otherwise this is a sign for me that Nextcloud is focusing more on larger customers (with support contracts) who have an environment in the background with which this function is not required.

More then 4 years are over and nothing was done. Hello!! Nextcloud!! What are the plans for this features.

**Comment by nickvergessen** [comment]
> I'm not a php programmer, but I think for the nextcloud experts it is not so hard to implement this functions (you can take a look at ownclouds implementation).

Feel free to find someone in https://help.nextcloud.com/c/nextcloud-freelancing/48 to get this feature in.

> This offends the community, which is also responsible for your success.

While I agree that the community is responsible for the success of Nextcloud, I don't think this issue is offending someone the community which (as you said yourself before) should be able to fix the issue. Especially since the NIST changed its recommendation and removed password expiration (ref https://blog.24by7security.com/unpacking-the-nist-password-requirements-in-2019) and now says it's not recommmended to expire passwords.

> ... otherwise this is a sign for me that Nextcloud is focusing more on larger customers (with support contracts) ...

This pays our salaries, that's just how it is and why you can use Nextcloud for free.

> Hello!! Nextcloud!! What are the plans for this features.

See https://github.com/nextcloud/server/blob/master/.github/CONTRIBUTING.md#contributing-to-source-code
The source code is in https://github.com/nextcloud/password_policy, send a pull request https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests yourself or as written above motivate others to do it for you.

**Comment by miguelns21** [comment]
Yes, I'm also interested.

**Comment by vince2010091** [comment]
There are 2 topics
Don't confuse "change at next login" with "password expiration policy"
When you give the user their first password, it is recommended to change it, and force the user to change it - regardless of expiration.

**Comment by Shen** [comment]
> Also as a company you may consider a Support Subscription from https://nextcloud.com/enterprise. This gives you direct access to Engineers and influence on our roadmap. 
> 

Is there no one with premium-subscription, who is interested to support/push this security(!)feature?
More than five years now. 😐

**Comment by netnut404** [comment]
is there any word on being able to force a password change?   I know people use resetting your password via email, and while that is a  work around that is ok for some   it really is a work around making you depend on a external account to be already setup a user properly  .  Alternatively It would be nice to have nextcloud be the first / only account for the user which  then set an easy password that must change at first login with a good password policy it will allow you to put all other accounts (including email) in a password database thus keeping all accounts secured

**Comment by ASLLR** [comment]
Hi everyone, i can see this post doesn't move since 1 year and this functionnality is always not implémented. Do you know in which version that can be integrated ? 

Thank in advance,

**Comment by RuudschMaHinda** [comment]
Just bumping in 2023 because there was an incident with one leaked password here.

**Comment by funktionierbar** [comment]
> Just bumping in 2023 because there was an incident with one leaked password here.

Same here, would be glad to have this feature!

**Comment by Salzorian** [comment]
I appreciate the work of the devs.
Please don't get me wrong.
This should be a default security feature?
thx in advanced.
